Refresh.Common/Extensions/RequestContextExtensions.cs

```diff
@@ -27,6 +27,43 @@ public static (int, int) GetPageData(this RequestContext context, int maxCount =
 
     [Pure]
     public static bool IsApi(this RequestContext context) => context.Url.AbsolutePath.StartsWith("/api/");
+
+    [Pure]
+    public static bool IsPatchworkVersionValid(this RequestContext context, int requiredMajor, int requiredMinor)
+    {
+        ReadOnlySpan<char> userAgent = context.RequestHeaders.Get("User-Agent").AsSpan();
+        if (userAgent.IsEmpty)
+            return false;
+
+        return IsPatchworkVersionValid(userAgent, requiredMajor, requiredMinor);
+    }
+
+    [Pure]
+    public static bool IsPatchworkVersionValid(ReadOnlySpan<char> userAgent, int requiredMajor, int requiredMinor)
+    {
+        // example: PatchworkLBP2 1.0
+        const string namePrefix = "PatchworkLBP";
+
+        // does the useragent even match patchwork's name?
+        if (!userAgent.StartsWith(namePrefix))
+            return false;
+        
+        // is the game version valid?
+        char gameVersion = userAgent[namePrefix.Length];
+        if (gameVersion is not '1' and not '2' and not '3' and not 'V')
+            return false;
+
+        // does the version string parse out?
+        if (!Version.TryParse(userAgent[(namePrefix.Length + 2)..], out Version? version))
```

Comment:
This parsing is flexible and should hold up if we want to add patch versions (e.g. 1.0.x)